bigcache.go

```diff
@@ -33,6 +33,79 @@ type cacheShard struct {
 	onRemove    func(wrappedEntry []byte)
 }
 
+// EntryInfo holds informations about entry in the cache
+type EntryInfo struct {
+	Key       string
+	Hash      uint64
+	Timestamp uint64
+	index     uint32
+}
+
+// EntryInfoIterator allows to iterate over entries in the cache
+type EntryInfoIterator struct {
+	cache        *BigCache
+	currentShard int
+	currentIndex int32
+	elements     []*EntryInfo
```

Comment:
This could be just slice of indexes `[]uin32`. We don't need all data during iteration.